### ISSUE TITLE
Incorrect mass calculated for sustainer stage in some cases

### ISSUE THREAD

**Issue by Patrykz94** [Noiredd/PEGAS#17]
This happens when you try to combine a upfg activation and separation of side boosters for a sustainer stage.

Since the mass calculations for sustainer stages are done around 5 seconds before the specified upfg actiavtion time (`upfgConvergenceDelay`), if you try to combine side boosters separation and activating the sustainer stage, pegas will include the mass of the still attached side boosters and treat it as extra fuel mass. This messes up the deltaV and burn duration of that stage.

Detecting those situations is quite easy, you can check if `ignition` is `FALSE` and `jettison` is `TRUE`. As for fixing this, maybe re-calculate the masses again after separation?

**Comment by Noiredd** [comment]
Yep, what PEGAS is not good at is doing many things at once. If you separate, wait a few seconds, separate something else, wait again, switch phases etc - everything should work well. But try to separate *and* switch guidance at the same time and it's hard to predict what will happen. I will look into this.

**Comment by Patrykz94** [comment]
I did some testing and the issue is still there (and another one is introduced).

I added a few lines of debug code that just print the fuel mass before and after recalculation.
![pegas](https://user-images.githubusercontent.com/13877840/32183255-b964e2b8-bd90-11e7-8602-a268e9c0ada3.PNG)
Looks like fuel is recalculated correctly but UPFG still seems to use the old mass.

And the other issue that happens when using this new code is that right after the upper stage separates, it flips the opposite direction to where it is supposed to be going. If I turn SAS on, manually turn it around and turn SAS off again, it locks into the right direction again.

Here is a [video example](https://drive.google.com/open?id=0Byb_VPJEOra9dWN0Z2dNN0dxLUU) of separating the boosters using `sequence` at 146 seconds, and activating UPFG 6 seconds later. Here, `jettison` is set to `FALSE`.

And here is a [video](https://drive.google.com/open?id=0Byb_VPJEOra9a2o1UE9SZHFIcjg) of activating UPFG at separation with the masses printed at the bottom. Settings are the same except `jettison` is `TRUE`, `waitBeforeJettison` is 1 second, no staging entry in sequence and `upfgActivation` is 146. Also had to move fairing jettisoning by 20 seconds so it doesn't interfere with anything.
